method.rb

```diff
@@ -0,0 +1,99 @@
+module Enumerable
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_arr = []
+    my_each do |num|
+      new_arr.push(num) if yield num
+    end
+    new_arr
+  end
+
+  def my_all?
+    stat = true
+    my_each do |num|
+      stat = false unless yield num
+    end
+    stat
+  end
+
+  def my_any?
+    stat = false
+    my_each do |num|
+      stat = true if yield num
+    end
+    stat
+  end
+
+  def my_none?
+    stat = true
+    my_each do |num|
+      stat = false if yield num
+    end
+    stat
+  end
+
+  def my_count(counter = size)
+    p_count = 0
+    if block_given?
+      my_each do |num|
+        p_count += 1 if yield num
+      end
+      counter = p_count
+    end
+    counter
+  end
+
+  def my_inject(acc = 0)
+    my_each do |num|
+      acc = yield(acc, num)
+    end
+    acc
+  end
+
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should work with symbols as arguments (:+) and perform the right operation.

- The use of my_each here is affecting your method's performance when calling it on a range.

Reply:
> * Your method should work with symbols as arguments (:+) and perform the right operation.
This kind of operation is only allowed with the reduce function not with the inject methods. Please check the documentation [inject](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject)